src/components/Button/index.tsx

```diff
@@ -118,6 +118,57 @@ type ButtonProps = (ButtonWithText | ChildrenProps) & {
     accessibilityLabel?: string;
 };
 
+type KeyboardShortcutComponentProps = Pick<ButtonProps, 'isDisabled' | 'isLoading' | 'onPress' | 'pressOnEnter' | 'allowBubble' | 'enterKeyEventListenerPriority'>;
+
+const accessibilityRoles: string[] = Object.values(CONST.ACCESSIBILITY_ROLE);
+
+const KeyboardShortcutComponent = memo(
+    ({isDisabled = false, isLoading = false, onPress = () => {}, pressOnEnter, allowBubble, enterKeyEventListenerPriority}: KeyboardShortcutComponentProps) => {
+        const isFocused = useRef(false);
+        const activeElementRole = useActiveElementRole();
+
+        const shouldDisableEnterShortcut = useMemo(() => accessibilityRoles.includes(activeElementRole ?? '') && activeElementRole !== CONST.ACCESSIBILITY_ROLE.TEXT, [activeElementRole]);
+
+        useFocusEffect(
+            useCallback(() => {
+                isFocused.current = true;
+
+                return () => {
+                    isFocused.current = false;
+                };
+            }, []),
+        );
+
+        const keyboardShortcutCallback = useCallback(
+            (event?: GestureResponderEvent | KeyboardEvent) => {
+                if (!validateSubmitShortcut(isDisabled, isLoading, event)) {
+                    return;
+                }
+                onPress();
+            },
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            [isDisabled, isLoading],
+        );
+
+        const config = useMemo(
+            () => ({
+                isActive: pressOnEnter && !shouldDisableEnterShortcut && isFocused.current,
+                shouldBubble: allowBubble,
+                priority: enterKeyEventListenerPriority,
+                shouldPreventDefault: false,
+            }),
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            [shouldDisableEnterShortcut],
+        );
+
+        useKeyboardShortcut(CONST.KEYBOARD_SHORTCUTS.ENTER, keyboardShortcutCallback, config);
```

Comment:
Seems like [regression](https://github.com/Expensify/App/issues/34239#issuecomment-1886239897) came from this change.
Let's try to fix this before being deploy blocker.

Reply:
@situchan PR created [here](https://github.com/Expensify/App/pull/34313). Will be ready for review shortly 👍

Reply:
ok, app deployed already and now this is deploy blocker.
I can do quick review if needed.
cc: @Julesssss